infoschema/slow_log.go

```diff
@@ -0,0 +1,182 @@
+package infoschema
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/execdetails"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/logutil"
+	log "github.com/sirupsen/logrus"
+)
+
+var slowQueryCols = []columnInfo{
+	{variable.SlowLogTimeStr, mysql.TypeDatetime, -1, 0, nil, nil},
+	{variable.SlowLogTxnStartTSStr, mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
+	{variable.SlowLogUserStr, mysql.TypeVarchar, 64, 0, nil, nil},
+	{variable.SlowLogConnIDStr, mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
+	{variable.SlowLogQueryTimeStr, mysql.TypeDouble, 22, 0, nil, nil},
+	{execdetails.ProcessTimeStr, mysql.TypeDouble, 22, 0, nil, nil},
+	{execdetails.WaitTimeStr, mysql.TypeDouble, 22, 0, nil, nil},
+	{execdetails.BackoffTimeStr, mysql.TypeDouble, 22, 0, nil, nil},
+	{execdetails.RequestCountStr, mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
+	{execdetails.TotalKeysStr, mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
+	{execdetails.ProcessedKeysStr, mysql.TypeLonglong, 20, mysql.UnsignedFlag, nil, nil},
+	{variable.SlowLogDBStr, mysql.TypeVarchar, 64, 0, nil, nil},
+	{variable.SlowLogIsInternalStr, mysql.TypeTiny, 1, 0, nil, nil},
+	{variable.SlowLogQuerySQLStr, mysql.TypeVarchar, 4096, 0, nil, nil},
+}
+
+func dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
+	rowsMap, err := parseSlowLogFile(ctx.GetSessionVars().SlowQueryFile)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var rows [][]types.Datum
+	for _, row := range rowsMap {
+		record := make([]types.Datum, 0, len(slowQueryCols))
+		for _, col := range slowQueryCols {
+			if v, ok := row[col.name]; ok {
+				record = append(record, v)
+			} else {
+				record = append(record, types.NewDatum(nil))
+			}
+		}
+		rows = append(rows, record)
+	}
+	return rows, nil
+}
+
+// TODO: Support parse multiple log-files.
+func parseSlowLogFile(filePath string) ([]map[string]types.Datum, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer func() {
+		if err = file.Close(); err != nil {
+			log.Error(err)
+		}
+	}()
+
+	return parseSlowLog(bufio.NewScanner(file))
+}
+
+// TODO: optimize for parse huge log-file.
+func parseSlowLog(scanner *bufio.Scanner) ([]map[string]types.Datum, error) {
+	rows := make([]map[string]types.Datum, 0)
+	rowMap := make(map[string]types.Datum, len(slowQueryCols))
+	startFlag := false
+	startPrefix := variable.SlowLogPrefixStr + variable.SlowLogTimeStr + variable.SlowLogSpaceMarkStr
```

Comment:
You can make startPrefix to be a global variable like `variable.SlowLogPrefixStr`